### ISSUE TITLE
Carla related query 

### ISSUE THREAD

**Issue by k-nayak** [Glutamat42/Ultra-Fast-Lane-Detection#3]
Hello @Glutamat42, i have a doubt on a problem i am working on. I am basically doing the same as your project to detect lanes in Carla in real time. My question is if it is possible to get the map (X,Y) coordinates and get a trajectory of the list of detected lane points for the car to drive on. I would offset the detected lane list to the middle of the lane to keep the car in the lane boundary. But i couldn't find any ways to transform the pixel coordinates to map coordinates yet. Any advice on this case would be very helpful. I am searching the carla forum as well, hoping to find some thing there. Thanks in advance.

**Comment by SoraMakes** [comment]
If i understand you correctly you want to convert a position on the camera image (x,y coordinate) to coordinates in the carla map.
In the other project related to this one we did it the other way around: https://github.com/Glutamat42/Carla-Lane-Detection-Dataset-Generation
Also maybe this issue might help you: https://github.com/cfzd/Ultra-Fast-Lane-Detection/issues/301#issuecomment-1114970734

**Comment by k-nayak** [comment]
![cam](https://user-images.githubusercontent.com/76793645/171423853-6f0afd14-1606-4eb8-a56b-6370e5a3c881.jpg)

basically, i want to get the XY map coordinates of the detected lane points so the car can use it as a trajectory. Is that the same type of thing you also did? In my case just using the center lane the green points and offsetting it to the center of the lane (this is the idea that i currently have)

**Comment by SoraMakes** [comment]
yep, exactly the opposite of what my partners did in the dataset generation project. You should be able to use it as a reference. Also because it wasn't my part of the project i can't help you that much with it. But as i also said in the issue i linked its a relatively common computer vision problem and you will find the formula for that conversion.

**Comment by k-nayak** [comment]
Thanks i will take a look into it and see how i can use it, if i am not wrong it's the [fast_lane_detection.py](https://github.com/Glutamat42/Carla-Lane-Detection-Dataset-Generation/blob/master/src/fast_lane_detection.py) where the world coordinate is converted to image coordinate. So when we say world coordinate in carla its the map coordinate right? the exact x y position on the map.

**Comment by SoraMakes** [comment]
i think so. i don't currently have the time to look it up myself

**Comment by k-nayak** [comment]
No worries, thanks for the input. I will check how I can do it for my case.
